helion/language/multicast_tensor.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from typing import NamedTuple
+
+import torch
+
+from .. import exc
+from . import _decorators
+
+if TYPE_CHECKING:
+    from .._compiler.type_propagation import TypeInfo
+    from .._compiler.variable_origin import Origin
+
+
+class MulticastTensor(NamedTuple):
```

Comment:
If we don't inherit from NamedTuple does this still work?

Reply:
Actually, no, we might want to keep the NamedTuple inheritation. `fake_fn` for hl ops is called during both type propagation and device_ir tracing. For type propagation multicast tensors are passed in the original MulticastTensor type and in device_ir tracing we call prepare_args to unpack the MulticastTensor into tuples before calling fake_fn. It is nicer make MulticastTensor a NamedTuple so that in both cases it is a tuple.

Reply:
Unless we find a better way to deal with MulticastTensor constructor in device_ir, and we don't need to unpack it into a tuple.